rando_modules/random_puzzles_minigames.py

```diff
@@ -0,0 +1,620 @@
+                        return "Red (3)"
+                    else:
+                        return "Blue (4)"
+                box_1 = random.randint(1, 4)
+                box_2 = random.randint(1, 4)
+                box_3 = random.randint(1, 4)
+                box_4 = random.randint(1, 4)
+                box_5 = random.randint(1, 4)
+                box_6 = random.randint(1, 4)
+                boxes_order = (
+                    (box_1 << 20)
+                  + (box_2 << 16)
+                  + (box_3 << 12)
+                  + (box_4 << 8)
+                  + (box_5 << 4)
+                  + box_6
+                )
+                spoilerlog_additions["GreenStationBoxes"] = (
+                    f"{_map_boxes(box_1)}, {_map_boxes(box_2)}, {_map_boxes(box_3)}, "
+                    f"{_map_boxes(box_4)}, {_map_boxes(box_5)}, {_map_boxes(box_6)}"
+                )
+            puzzle_minigame_list.append((puzzle.get_key(), boxes_order))
+
+        # Deep Jungle Push Blocks: Initial positions
+        elif puzzle.name in [
+            "DeepJunglePushBlocks1","DeepJunglePushBlocks2",
+            "DeepJunglePushBlocks3","DeepJunglePushBlocks4"
+        ]:
+            if not random_puzzles:
+                positions_encoded = puzzle.default_value
+            else:
+                positions_encoded, deepjungle_blocked_positions = _deepjungle_pushblock_positions(
+                    puzzle.name,
+                    deepjungle_blocked_positions
+                )
+            puzzle_minigame_list.append((puzzle.get_key(), positions_encoded))
+
+        # Ultra Hammer room Push Blocks: Initial positions
+        elif puzzle.name == "UltraHammerPushBlocks":
+            if not random_puzzles:
+                positions_encoded = puzzle.default_value
+            else:
+                positions_encoded = _random_pushblock_positions(
+                    num_blocks = 2,
+                    min_x = 5,
+                    max_x = 14,
+                    min_z = 0,
+                    max_z = 4,
+                    disallowed_positions = [(14, 1)]
+                )
+            puzzle_minigame_list.append((puzzle.get_key(), positions_encoded))
+
+        # Lava Dam Push Blocks: Initial positions
+        elif puzzle.name == "LavaDamPushBlocks":
+            if not random_puzzles:
+                positions_encoded = puzzle.default_value
+            else:
+                positions_encoded = _lavadam_pushblock_positions()
+            puzzle_minigame_list.append((puzzle.get_key(), positions_encoded))
+
+        # Flower Fields Three Tree: Correct hit sequence
+        elif puzzle.name == "FlowerFieldsThreeTrees":
+            if not random_puzzles:
+                sequence_encoded = puzzle.default_value
+                spoilerlog_additions["FlowerFieldsThreeTrees"] = "Middle, Right, Left"
+            else:
+                def _map_tree(tree_id: int) -> str:
+                    if tree_id == 1:
```

Comment:
Shouldn't it be if tree_id ==0?

Reply:
Doesn't really matter, as long as the mod can handle whatever we feed it. For this one I opted for just numbering them naturally, like "tree to hit number 1, number 2, number 3"